libpod/image.go

```diff
@@ -50,29 +55,116 @@ type CopyOptions struct {
+	}
+
+	tagged, istagged := imgRef.(reference.NamedTagged)
+	tag := "latest"
+	if istagged {
+		tag = tagged.Tag()
+	}
+	hasDomain := true
+	registry := reference.Domain(imgRef.(reference.Named))
+	if registry == "" {
+		hasDomain = false
+	}
+	imageName := reference.Path(imgRef.(reference.Named))
+	pImage := imagePullStruct{
+		imageName,
+		tag,
+		registry,
+		hasDomain,
+		DefaultRegistry,
+	}
+	if pImage.hasRegistry {
+		// If input has a registry, we have to assume they included an image
+		// name but maybe not a tag
+		pullRef, err := alltransports.ParseImageName(pImage.returnFQName())
+		if err != nil {
+			return registries, errors.Errorf(imageError)
+		}
+		registries = append(registries, pullRef)
+	} else {
+		// No registry means we check the globals registries configuration file
+		// and assemble a list of candidate sources to try
+		registryConfigPath := ""
+		envOverride := os.Getenv("REGISTRIES_CONFIG_PATH")
+		if len(envOverride) > 0 {
+			registryConfigPath = envOverride
+		}
+		searchRegistries, err := sysregistries.GetRegistries(&types.SystemContext{SystemRegistriesConfPath: registryConfigPath})
+		if err != nil {
+			fmt.Println(err)
+			return registries, errors.Errorf("Unable to parse the registries.conf file and"+
+				" the image name '%s' is incomplete.", imageName)
+		}
+		for _, searchRegistry := range searchRegistries {
+			pImage.registry = searchRegistry
+			pullRef, err := alltransports.ParseImageName(pImage.returnFQName())
+			if err != nil {
+				return registries, errors.Errorf(imageError)
+			}
+			registries = append(registries, pullRef)
+		}
+	}
+	return registries, nil
+}
+
+// PullImage copies the image from the source to the destination
+// configured
+func (r *Runtime) PullImage(imgName string, allTags bool, reportWriter io.Writer) error {
+	//func PullImage(store storage.Store, imgName string, allTags, quiet bool, sc *types.SystemContext) error {
+	// An eventual array of imagereferences to pull on
+	var fqregistries []types.ImageReference
+	srcRef, err := alltransports.ParseImageName(imgName)
+	if err != nil {
+		fqregistries, err = getRegistriesToTry(imgName)
+		if err != nil {
+			fmt.Println(err)
+		}
+	} else {
+		fqregistries = append(fqregistries, srcRef)
+	}
+
+	for _, pullRef := range fqregistries {
+		result := pullImage(imgName, allTags, reportWriter, r, pullRef)
+		if result == nil {
+			return nil
+		}
+	}
+	return errors.Errorf("Unable to pull image %s", imgName)
+}
+
 // PullImage pulls an image from configured registries
 // By default, only the latest tag (or a specific tag if requested) will be
 // pulled. If allTags is true, all tags for the requested image will be pulled.
 // Signature validation will be performed if the Runtime has been appropriately
 // configured
-func (r *Runtime) PullImage(imgName string, allTags bool, reportWriter io.Writer) error {
+func pullImage(imgName string, allTags bool, reportWriter io.Writer, r *Runtime, srcRef types.ImageReference) error {
```

Comment:
I think this should be a method on runtime, not accept it as an argument, for consistency - so `func (r *Runtime) pullImage(...)` instead

Reply:
fixed